compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -496,11 +496,6 @@ OMR::Z::TreeEvaluator::l2aEvaluator(TR::Node * node, TR::CodeGenerator * cg)
 
    node->setRegister(source);
 
-
-   if (firstChild->containsCompressionSequence())
```

Comment:
This is a functional fix as I hit the assertions when we attempted to reduce the reference count of child.  I believe we also evaluate the children of firstChild when we call evaluate on it.